src/compiler/crystal/tools/playground/server.cr

```diff
@@ -490,7 +490,7 @@ module Crystal::Playground
         origin = context.request.headers["Origin"]
         if !accept_request?(origin)
           Log.warn { "Invalid Request Origin: #{origin}" }
-          ws.close "Invalid Request Origin"
+          ws.close 1008, "Invalid Request Origin"
```

Comment:
Should these numeric codes be turned into an enum?  This would also allow the type restrictions to be a bit more clear on what the possible values are.

It could also allow for some method on the enum member that would return its byte representation.

Reply:
No, because there's a set of predefined ones *and* a range of possible undefined values (4000-4999). It'd turn into `WebSocket::CloseCode | Int` and it's probably not worth it.

There could be constants for the ~10 predefined codes, but otherwise I don't think there's much benefit.

Reply:
Fair enough.  FWIW it could be done similar to `HTTP::Status`.  Where the known ones are defined, but you can still do `HTTP::Status.new 800`.

In which case it could validate that the custom value is within the required range.  This would keep the possible types to the enum itself.  It would also provide validation in what gets sent is valid at a cost of more verbosity.  `WebSocket::CloseCode.new(4500)` versus `4500`.

Reply:
In the future I would like to disallow Enum with non-valid values

Reply:
If not an enum, maybe at least extract them to constants?